content/en/docs/reference/controllers/horizontal-pod-autoscaler.md

```diff
@@ -0,0 +1,163 @@
+---
+title: Horizontal Pod Autoscaler
+content_template: templates/concept
+---
+
+{{% capture overview %}}
+
+The Horizontal Pod Autoscaler automatically scales the number of Pods
+in a {{< glossary_tooltip text="Deployment" term_id="deployment" >}}, {{< glossary_tooltip text="StatefulSet" term_id="statefulset" >}} or other scalable resource.
+This controller scales the number of Pods based on observed CPU utilization (or, with
+[custom metrics](https://git.k8s.io/community/contributors/design-proposals/instrumentation/custom-metrics-api.md)
+support, on some other application-provided metrics).
+
+{{% /capture %}}
+{{% capture body %}}
+
+## Automatic scaling
+
+The controller periodically adjusts the number of replicas in Deployment (or ReplicationController),
+so as to to match the observed average resource utilization to the target specified by user.
+
+![Horizontal Pod Autoscaler diagram](/images/docs/horizontal-pod-autoscaler.svg)
+
+The Horizontal Pod Autoscaler is implemented as a control loop, with a period controlled
+by the controller manager's `--horizontal-pod-autoscaler-sync-period` flag.
+The default synchronization period is 15 seconds.
+
+During each period, the controller manager queries the resource utilization against the
+metrics specified in each HorizontalPodAutoscaler definition.  The controller manager
+obtains the metrics from either the resource metrics API (for per-pod resource metrics),
+or the custom metrics API (for all other metrics).
+
+* For per-pod resource metrics (like CPU), the controller fetches the metrics
+  from the resource metrics API for each pod targeted by the HorizontalPodAutoscaler.
+  Then, if a target utilization value is set, the controller calculates the utilization
+  value as a percentage of the equivalent resource request on the containers in
+  each pod.  If a target raw value is set, the raw metric values are used directly.
+  The controller then takes the mean of the utilization or the raw value (depending on the type
+  of target specified) across all targeted pods, and produces a ratio used to scale
+  the number of desired replicas.
+
+  If some of the Pod's containers do not have the relevant resource request set,
+  CPU utilization for the pod will not be defined and the autoscaler will
+  not take any action for that metric. See [algorithm
+  details](#algorithm-details) for more information about
+  how the autoscaling algorithm works.
+
+* For per-pod custom metrics, the controller functions similarly to per-pod resource metrics,
```

Comment:
I've seen a number of questions about what "used" means with regards to resource metrics on Slack and the mailing lists. It's probably a good idea to put a note in here explaining what "raw values, not utilization values" means. Perhaps something like "Resource utilization values are calculated from the Pod and Container resource request and limit amounts. Raw utilization values, on the other hand, are queried from the Node's underlying operating system."